### DISCUSSION TITLE
Does typed-document-node replace some typescript-react-apollo features?

### DISCUSSION THREAD

**Discussion by miller-productions** [dotansimha/graphql-code-generator#5201] [Q&A] [answered]
Hi there,

I hve just come across `typed-document-node` and wondering if it replaces some features from `typescript-react-apollo`?

For example, in [one of the code examples](https://github.com/dotansimha/graphql-typed-document-node/blob/master/examples/apollo-client-3/index.ts) we see that the approach is to work with `useQuery` from `@apollo/client`. 

Would this replace the approach of working [withHooks](https://graphql-code-generator.com/docs/plugins/typescript-react-apollo#with-react-hooks) from `typescript-react-apollo`?

**Comment by miller-productions** [comment, marked as answer]
Reading this discussion #4580 , I suspect the answer is yes: you probably don't need `typescript-react-apollo` if you are using `typed-document-node`.

@dotansimha mentions it [here](https://github.com/dotansimha/graphql-code-generator/discussions/4580#discussioncomment-86689)

Did I interpret that correctly?

**Reply by dotansimha** [reply]
You are right, the question is yes. But note that it depends on adoption in actual libraries (or patches we provide here: https://github.com/dotansimha/graphql-typed-document-node). 

There are limitations, since we can't offer new features on the generated code, and it all depends on implementing clients. 

In terms of development, we are still maintaining `typescript-react-apollo` (and others) and we'll keep doing that, since it's heavily used.

**Reply by miller-productions** [reply]
Thanks Dotan,

So, if a library that you are using has [built-in support](https://github.com/dotansimha/graphql-typed-document-node#built-in-support) then you should be ok to start using `typed-document-node`. 

However, if the library you are using doesn't yet have built-in support, then you will be required to [patch](https://github.com/dotansimha/graphql-typed-document-node#with-patch) in order to use `typed-document-node`.

That is helpful. Much appreciated.
